lib/whitehall/not_quite_as_fake_search.rb

```diff
@@ -185,13 +185,15 @@ def filter_by_boolean_field(field, desired_field_value, document_hashes)
         document_hashes.select { |document_hash| document_hash[field] == desired_boolean }
       end
 
-      def filter_by_simple_field(field, desired_field_values, document_hashes)
+      def filter_by_simple_field(field, desired_field_values, document_hashes, announcements_search = false)
         document_hashes.select do |document_hash|
-          if field == "organisations"
-            value = organisation_slugs(document_hash["organisations"])
-          else
-            value = document_hash.fetch(field, [])
-          end
+          value =
+            if field == "organisations" && announcements_search
+              organisation_slugs(document_hash.fetch("organisations", []))
+            else
+              document_hash.fetch(field, [])
+            end
```

Comment:
Was this conditional only applied on announcements searches before? I just notice that it will no longer ever return true if you filtered a publication by organisations for instance.

Reply:
We need to treat organisations differently for announcements as the '/search' endpoint returns them as an array of hashes, while '/advanced-search' returns them as an array of slugs.  This logic should cater for non-announcement documents as `announcements_search` will be set to false in these cases and with fall through to the standard 'fetch' from the document hash.